docs: add CORS documentation to 'Configuration Overview' page (#1477)
cube-js/cube.js: docs/Configuration/Overview.md

menuOrder: 1
---

Cube.js is designed to work with different configuration sources. There are two

> main ways you can set configuration options; via [a configuration

Reviewer: ```suggestion
ways you can set configuration options; via [a configuration
```

"Main" is redundant here, right?

## Configuration Precedence

In Cube.js, environment variables take precedence over values specified in

> `cube.js`. In production scenarios, it allows changing configuration by simply

Reviewer: ```suggestion
`cube.js`. In production scenarios, it allows changing configuration by
```

How simple it is, depends on the tools used, right? But won't it be more concise to omit "simply" here?

switching environment variables, which is generally safer, faster and easier
than re-deploying the codebase.

## Development Mode

Cube.js can be run in an insecure, development mode by setting the
`CUBEJS_DEV_MODE` environment variable to `true`. Putting Cube.js in development
mode does the following:

- Disables authentication checks
- Allows another log level to be set (`trace`)
- Enables the Playground on `http://localhost:4000`
- Uses `memory` instead of `redis` as the default cache/queue engine
- Logs incorrect/invalid configuration for `externalRefresh` /`waitForRenew`
  instead of throwing errors

## Configuring CORS

The Cube.js REST API supports Cross-Origin Resource Sharing (CORS) through the
[Express CORS middleware][link-express-cors]. The middleware can be configured

> by setting [any available options][link-express-cors-opts] in the `cube.js`

Reviewer: ```suggestion
by setting [relevant options][link-express-cors-opts] in the `cube.js`
```

Maybe saying "any" is quite misleading here?

configuration file:

[link-express-cors]: https://expressjs.com/en/resources/middleware/cors.html
[link-express-cors-opts]:
  https://expressjs.com/en/resources/middleware/cors.html#configuration-options

```javascript
module.exports = {
  http: {
    cors: {
      allowedHeaders: 'authorization,content-type,x-request-id', // By default
      exposedHeaders: 'content-type',
      origin: 'https://myapp.com',
    },
  },
};
```

## Migrating from Express to Docker

Since [`v0.23`][link-v-023-release], Cube.js CLI uses the `docker` template
instead of `express` as a default for project creation, and it is the

> recommended route for production. To migrate you should move all of your Cube.js
> dependencies in `package.json` to `devDependencies` and leave all of your

Reviewer: ```suggestion
recommended route for production. To migrate you should move all Cube.js
dependencies in `package.json` to `devDependencies` and leave
```

Just curious, is it correct from the language standpoint to remove these?
